lib/index.js

```diff
@@ -73,15 +70,13 @@ internals.implementation = function (server, options) {
 
     server.state(settings.cookie, cookieOptions);
 
-    const decoration = function (request) {
+    const decoration = (request) => {
 
-        const CookieAuth = function () {
+        class CookieAuth {
```

Comment:
Probably the class can be extracted and `request` attached to it through the constructor.

Reply:
Done, also had to attach the settings.